### PR TITLE
Improve the error for the drop table in system database

### LINKED ISSUE #2841: Improve the error for the drop table in system database
**Summary**

MySQL client:
```
mysql> use system;
Database changed
mysql> drop table xx;
ERROR 1105 (HY000): Code: 3, displayText = system.
```

server logs:
```
[2021-11-17T05:29:13Z ERROR databend_query::servers::mysql::writers::query_result_writer] OnQuery Error: Code: 3, displayText = system.
    
       0: common_exception::exception::ErrorCode::UnknownDatabase
                 at common/exception/src/exception.rs:110:77
       1: common_meta_raft_store::state_machine::sm::StateMachine::get_db_id::{{closure}}::{{closure}}
                 at common/meta/raft-store/src/state_machine/sm.rs:596:28
       2: core::option::Option<T>::ok_or_else
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/option.rs:942:25
       3: common_meta_raft_store::state_machine::sm::StateMachine::get_db_id::{{closure}}
                 at common/meta/raft-store/src/state_machine/sm.rs:593:23
       4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
       5: common_meta_raft_store::state_machine::sm::StateMachine::apply_cmd::{{closure}}::{{closure}}
                 at common/meta/raft-store/src/state_machine/sm.rs:420:29
       6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
       7: common_meta_raft_store::state_machine::sm::StateMachine::apply_cmd::{{closure}}
                 at common/meta/raft-store/src/state_machine/sm.rs:295:5
       8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
       9: common_meta_embedded::meta_api_impl::<impl common_meta_api::meta_api::MetaApi for common_meta_embedded::meta_embedded::MetaEmbedded>::drop_table::{{closure}}
                 at common/meta/embedded/src/meta_api_impl.rs:168:19
      10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
      11: <core::pin::Pin<P> as core::future::future::Future>::poll
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/future.rs:119:9
```

### PR DESCRIPTION
**Pull request by miamia0**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary
pre-check the whether  `dbname` of the deleted table belongs to the system database.

## Changelog
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #2841

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/meta/raft-store/src/state_machine/sm.rs [hunk 1]
```diff
@@ -417,6 +417,12 @@ impl StateMachine {
                 ref db_name,
                 ref table_name,
             } => {
+                if db_name == "system" {
```
Comment (BohuTANG) [inline common/meta/raft-store/src/state_machine/sm.rs:420]: Em, we should throw the error in query/catalog: https://github.com/datafuselabs/databend/blob/9c2e1d606688bbac5134eaaee33dfb3b798dfb8d/query/src/catalogs/impls/catalog/system_catalog.rs#L158

But it doesn't, it need to address why

Comment (miamia0) [reply common/meta/raft-store/src/state_machine/sm.rs]: fixed :)

**Comment by BohuTANG** [comment]
Great, it would be better to add a case to stateless test in:
https://github.com/datafuselabs/databend/blob/9c2e1d606688bbac5134eaaee33dfb3b798dfb8d/tests/suites/0_stateless/05_0000_ddl_drop_tables.sql#L1-L7

Like:
```
USE system;
DROP TABLE t; -- {Your ErrorCode} 
```

**Pull request merged**